src/main/groovy/life/qbic/samplestatus/reporter/services/RealLimsQueryService.groovy

```diff
@@ -0,0 +1,145 @@
+     * and maps it to the parameter <code>serverTimeout</code>.
+     *
+     * @param openbisUser the openBIS user id
+     * @param openbisPassword the openBIS user password
+     * @param applicationServerUrl the openBIS application server url
+     * @param serverTimeout the server connection timeout
+     */
+    RealLimsQueryService(@Value('${service.openbis.user}') String openbisUser,
+                         @Value('${service.openbis.password}') String openbisPassword,
+                         @Value('${service.openbis.api-server-url}') String applicationServerUrl,
+                         @Value('${service.openbis.server-timeout}') Integer serverTimeout) {
+        this.openBisApplicationServerApi = HttpInvokerUtils.createServiceStub(
+                IApplicationServerApi.class,
+                applicationServerUrl + IApplicationServerApi.SERVICE_URL, serverTimeout)
+        sessionToken = this.openBisApplicationServerApi.login(openbisUser, openbisPassword)
+    }
+
+    /**
+     * <p>This method is called by the Spring framework, after an instance of this class
+     * has been created.</p>
+     * <p>We just check, if the authentication against openBIS has been successful
+     * which is the case, when there is a session token present.</p>
+     */
+    @PostConstruct
+    void init() {
+        if (!sessionToken) {
+            throw new AuthenticationException("Authentication against LIMS service failed.")
+        }
+    }
+
+    /**
+     * {@InheritDocs}
+     */
+    @Override
+    List<SampleUpdate> getUpdatedSamples(Instant updatedSince) {
+        SampleSearchCriteria criteria = new SampleSearchCriteria()
+        // we make sure that the barcode is set, otherwise the sample is of no interest to us
+        criteria.withProperty("QBIC_BARCODE").thatContains("Q")
+        // only fetch latest samples
+        criteria.withModificationDate().thatIsLaterThanOrEqualTo(Date.from(updatedSince))
+ 
+        // we need to fetch properties, as the sample status (and barcode) is contained therein
+        SampleFetchOptions fetchOptions = new SampleFetchOptions()
+        fetchOptions.withProperties()
+
+        SearchResult<Sample> result = v3.searchSamples(sessionToken, criteria, fetchOptions)
+        List<SampleUpdate> sampleUpdates = result.getObjects().stream().map( sample -> createSampleUpdate(sample) ).collect( Collectors.toList() )
+
+        return sampleUpdates
+    }
+
+    public static SampleUpdate createSampleUpdate(Sample limsSample) {
+
+        Map<String,String> properties = limsSample.getProperties()
+        String sampleBarcode = properties.get("QBIC_BARCODE")
+
+        life.qbic.samplestatus.reporter.Sample sample = new life.qbic.samplestatus.reporter.Sample(sampleBarcode)
+
+        Date modificationDate = sample.getModificationDate()
+        String updatedStatus = mapSampleStatus(properties.get("SAMPLE_STATUS"))
```

Comment:
I think we need to indroduce some exception handling. What do we do if the status entry is empty? 

We could introduce an own RuntimeException subclass, that is thrown when this field is empty with a descriptive message